test/annotation_dbg_info_drop.ll

```diff
@@ -0,0 +1,149 @@
+  %3 = bitcast %"class._ZTSZZ4mainENK3$_0clERN2cl4sycl7handlerEEUlvE_.anon"* %0 to i8*, !dbg !26
+  call void @llvm.lifetime.end.p0i8(i64 1, i8* %3) #5, !dbg !26
+  ret void, !dbg !26
+}
+
+; Function Attrs: argmemonly nounwind
+declare void @llvm.lifetime.start.p0i8(i64 immarg, i8* nocapture) #1
+
+; Function Attrs: nounwind readnone speculatable
+declare void @llvm.dbg.declare(metadata, metadata, metadata) #2
+
+; CHECK-SPIRV: {{[0-9]+}} Name [[STR_ID:[0-9]+]] ".str"
+; CHECK-SPIRV: {{[0-9]+}} Name [[STR1_ID:[0-9]+]] ".str.1"
+; CHECK-SPIRV: {{[0-9]+}} Name [[FUNC_ID:[0-9]+]] "_ZZZ4mainENK3$_0clERN2cl4sycl7handlerEENKUlvE_clEv"
+; CHECK-SPIRV: {{[0-9]+}} Name [[BUF_ID:[0-9]+]] "Buf"
+; CHECK-SPIRV: {{[0-9]+}} Name [[F_ID:[0-9]+]] "_Z1fv"
+
+; CHECK-SPIRV: {{[0-9]+}} Function {{[0-9]+}} [[FUNC_ID]]
+; -- 1. In SPIR-V there shouldn't be DebugScope instruction in variables allocation block.
+; CHECK-SPIRV:      2 Label
+; CHECK-SPIRV-NEXT: 4 Variable
+; CHECK-SPIRV-NEXT: 4 Variable
+; -- 2. DebugScope instruction should be added after annotation intrinsic call processing.
+; CHECK-SPIRV:      4 Bitcast {{[0-9 ]+}} [[BUF_ID]]{{[[:space:]]+}}5 ExtInst {{[0-9 ]+}} DebugNoScope
+; CHECK-SPIRV-NEXT: 6 InBoundsPtrAccessChain 
+; CHECK-SPIRV-NEXT: 6 InBoundsPtrAccessChain
+; CHECK-SPIRV-NEXT: 6 ExtInst {{[0-9 ]+}} DebugScope
+; CHECK-SPIRV-NEXT: 4 Line
+; CHECK-SPIRV-NEXT: 4 FunctionCall {{[0-9 ]+}} [[F_ID]]
+
+; Function Attrs: inlinehint nounwind
+define internal spir_func void @"_ZZZ4mainENK3$_0clERN2cl4sycl7handlerEENKUlvE_clEv"(%"class._ZTSZZ4mainENK3$_0clERN2cl4sycl7handlerEEUlvE_.anon" addrspace(4)* %this) #3 align 2 !dbg !29 {
+entry:
+  %this.addr = alloca %"class._ZTSZZ4mainENK3$_0clERN2cl4sycl7handlerEEUlvE_.anon" addrspace(4)*, align 8
+  %Buf = alloca [1 x i8], align 1
+  store %"class._ZTSZZ4mainENK3$_0clERN2cl4sycl7handlerEEUlvE_.anon" addrspace(4)* %this, %"class._ZTSZZ4mainENK3$_0clERN2cl4sycl7handlerEEUlvE_.anon" addrspace(4)** %this.addr, align 8, !tbaa !34
+  call void @llvm.dbg.declare(metadata %"class._ZTSZZ4mainENK3$_0clERN2cl4sycl7handlerEEUlvE_.anon" addrspace(4)** %this.addr, metadata !31, metadata !DIExpression()), !dbg !38
+  %this1 = load %"class._ZTSZZ4mainENK3$_0clERN2cl4sycl7handlerEEUlvE_.anon" addrspace(4)*, %"class._ZTSZZ4mainENK3$_0clERN2cl4sycl7handlerEEUlvE_.anon" addrspace(4)** %this.addr, align 8
+  %0 = bitcast [1 x i8]* %Buf to i8*, !dbg !39
+  call void @llvm.lifetime.start.p0i8(i64 1, i8* %0) #5, !dbg !39
+  call void @llvm.dbg.declare(metadata [1 x i8]* %Buf, metadata !33, metadata !DIExpression()), !dbg !40
+  %Buf2 = bitcast [1 x i8]* %Buf to i8*, !dbg !39
+  call void @llvm.var.annotation(i8* %Buf2, i8* getelementptr inbounds ([25 x i8], [25 x i8]* @.str, i32 0, i32 0), i8* getelementptr inbounds ([15 x i8], [15 x i8]* @.str.1, i32 0, i32 0), i32 13), !dbg !39
```

Comment:
Please add checks for `@llvm.ptr.annotation` and `@llvm.annotation`.

Reply:
Updated the test.